Design/indexing_rollout_notes.md

```diff
@@ -0,0 +1,209 @@
+}
+```
+
+## Duplicate spectrum numbers
+
+Currently quite a few algorithms or tests use or create workspaces with non-unique spectrum numbers.
+Ultimately we want to forbid this, but it is not yet clear if we should attempt to clean this up right away, or wait until `IndexInfo` has developed into a more stable state.
+
+
+## Standards
+
+#### Spectrum numbers starting at 0
+
+Currently quite a few algorithms that create workspaces where spectrum numbers do not start at 1 but at 0.
+This is non standard.
+Should this be cleaned up as part of this refactoring effort?
+
+Examples:
+
+- `AppendSpectra`
+
+
+## MatrixWorkspace
+
+There is a considerable number of methods in `MatrixWorkspace` that could potentially be replaced or removed during this refactoring effort.
+The detailed strategy should depend on how and where these methods are used.
+
+```cpp
+class MatrixWorkspace {
+public:
+  void updateSpectraUsing(const SpectrumDetectorMapping &map);
+  void rebuildSpectraMapping(const bool includeMonitors);
+  spec2index_map getSpectrumToWorkspaceIndexMap() const;
+  detid2index_map getDetectorIDToWorkspaceIndexMap(bool throwIfMultipleDets) const;
+  std::vector<size_t> getDetectorIDToWorkspaceIndexVector(detid_t &offset, bool throwIfMultipleDets) const;
+  std::vector<size_t> getSpectrumToWorkspaceIndexVector(specnum_t &offset) const;
+  std::vector<size_t> getIndicesFromSpectra(const std::vector<specnum_t> &spectraList) const;
+  size_t getIndexFromSpectrumNumber(const specnum_t specNo) const;
+  std::vector<size_t> getIndicesFromDetectorIDs(const std::vector<detid_t> &detIdList) const;
+  std::vector<specnum_t> getSpectraFromDetectorIDs(const std::vector<detid_t> &detIdList) const;
+};
+```
+
+
+## Notes about individual algorithms
+
+The following is a more or less random selection/collection about specific algorithms or classes and how the could or should change.
+
+#### SumSpectra
+
+Uses smallest spectrum number in sum as spectrum number for output.
+This does not seem to make much sense, maybe it is just an arbitrary requirement by an individual piece of client code.
+It would probably be easier and cleaner to simply remove special handling of spectrum numbers from `SumSpectra` and adapt the client code in question.
+
+#### MDWorkspace
+
+`ConvertToDiffractionMDWorkspace::convertEventList` takes an `EventList` and its its events to an `MDWorkspace`.
+This procedure requires the position of the `EventList`.
+Currently this is determined by the detector IDs stored in the `EventList`.
+A different mechanism is required here, probably with a change in the function signature.
+
+#### SpectrumDetectorMapping
+
+This class is used in a few algorithms, e.g., `SofQWNormalisedPolygon`, to set/update the detector IDs in a workspace via `MatrixWorkspace::updateSpectraUsing`.
+Probably this functionality can simply be replaced by basic `IndexInfo` functionality.
+Ultimately, it might be possible to remove the class `SpectrumDetectorMapping`.
```

Comment:
Agreed. The `IndexInfo`needs to be the unique, consistent and unambiguous source of this information.